### PR TITLE
Apr 1569 oversight overall outcome

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/OversightController.cs [hunk 1]
```diff
@@ -1,8 +1,10 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 using MediatR;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using NPOI.HPSF;
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/OversightController.cs:7]: Is this needed?

File: src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/OversightHandlerTests/OversightDetailsHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                ApplicationReferenceNumber = "APR000111",
+                OversightStatus = "Successful",
+                ApplicationDeterminedDate = DateTime.Today
+            };
+
+            ApplyRepository.Setup(r => r.GetOversightDetails(applicationId)).ReturnsAsync(applicationDetails);
+
+            var result = _handler.Handle(new GetOversightDetailsRequest(applicationId), new CancellationToken()).GetAwaiter().GetResult();
+
+            Assert.AreEqual(applicationId, result.ApplicationId);
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/OversightHandlerTests/OversightDetailsHandlerTests.cs:49]: This assertion doesn't feel sufficient, an IsSame on the result and applicationDetails would be a better check.

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/OversightControllerTests.cs [hunk 3]
```diff
@@ -97,5 +97,40 @@ public async Task Check_pending_results_are_as_expected()
...
+
+            _mediator
+                .Setup(x => x.Send(It.IsAny<GetOversightDetailsRequest>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(oversight);
+
+            var actualResult = await _controller.OversightDetails(applicationId);
+
+            var returnedOversight = actualResult.Value;
+
+            Assert.AreEqual(oversight.Id, returnedOversight.Id);
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.InternalApi.UnitTests/OversightControllerTests.cs:126]: In contract to my previous comment, these checks are probably overkill.  IsSame would be better and more appropriate to the code in the controller.

File: src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/OversightHandlerTests/OversightDetailsHandlerTests.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.Application.UnitTests/Handlers/OversightHandlerTests/OversightDetailsHandlerTests.cs]: Mike I've done the changes but the GIT has got to a state it won't allow me to re-review.  Can I mark your comments as resolved and see if allows things to move on?

**Pull request closed without merging**